File: programmers/체육복.py
def solution(n, lost, reserve):
    count = n - len(lost)

    for i in lost:
        if i in reserve:
            lost.remove(i)
            reserve.remove(i)
            count += 1


    for i in lost:
        if i-1 in reserve:
            count += 1
            reserve.remove(i-1)
            continue
        if i+1 in reserve:
            count += 1
            reserve.remove(i+1)
            continue
        if i in reserve:
            count += 1
            reserve.remove(i)
            continue

    return count

File: programmers/체육복.js
function solution(n, lost, reserve) {
  lost.sort();
  reserve.sort();
  let _lost = lost.filter((l) => !reserve.includes(l));
  let _reserve = reserve.filter((r) => !lost.includes(r));
  let store = [];

  _lost.forEach((l) => {
    if (_reserve.includes(l - 1)) {
      const idx = _reserve.indexOf(l - 1);
      _reserve.splice(idx, 1);
      store.push(l);
    } else if (_reserve.includes(l + 1)) {
      const idx = _reserve.indexOf(l + 1);
      _reserve.splice(idx, 1);
      store.push(l);
    }
  });

  let result = _lost.filter((item) => !store.includes(item));

  return n - result.length;
}
